viadot/sources/s3.py

```diff
@@ -1,266 +1,132 @@
-            # Show a useful error message.
-            if recursive:
-                msg = f"At least one file in '{to_path}' already exists. Specify `overwrite=True` to overwrite."  # noqa
-            else:
-                msg = f"The file '{to_path}' already exists. Specify `overwrite=True` to overwrite."
-            raise FileExistsError(msg)
-
-        self.logger.info(
-            f"Successfully uploaded file(s) from '{from_path}' to '{to_path}'."
-        )
-
-    def exists(self, path: str = None) -> bool:
+    def exists(self, path: str) -> bool:
         """
-        Check if a location exists in Azure Data Lake.
-
+        Check if a location exists in S3.
         Args:
             path (str): The path to check. Can be a file or a directory.
-
-        Example:
-        ```python
-        from viadot.sources import AzureDataLake
-
-        lake = AzureDataLake(gen=1)
-        lake.exists("tests/test.csv")
-        ```
-
         Returns:
             bool: Whether the paths exists.
         """
-        path = path or self.path
-        return self.fs.exists(path)
-
-    def download(
-        self,
-        to_path: str,
-        from_path: str = None,
-        recursive: bool = False,
-        overwrite: bool = True,
-    ) -> None:
-
-        if overwrite is False:
-            raise NotImplemented(
-                "Currently, only the default behavior (overwrite) is available."
-            )
-
-        from_path = from_path or self.path
-        self.fs.download(rpath=from_path, lpath=to_path, recursive=recursive)
-
-    @add_viadot_metadata_columns
-    def to_df(
-        self,
-        path: str = None,
-        sep: str = "\t",
-        quoting: int = 0,
-        lineterminator: str = None,
-        error_bad_lines: bool = None,
-    ):
-        if quoting is None:
-            quoting = 0
-
-        path = path or self.path
-        url = os.path.join(self.base_url, path)
+        return wr.s3.does_object_exist(boto3_session=self.session, path=path)
 
-        if url.endswith(".csv"):
-            df = pd.read_csv(
-                url,
-                storage_options=self.storage_options,
-                sep=sep,
-                quoting=quoting,
-                lineterminator=lineterminator,
-                error_bad_lines=error_bad_lines,
-            )
-        elif url.endswith(".parquet"):
-            df = pd.read_parquet(url, storage_options=self.storage_options)
-        else:
-            raise ValueError("Only CSV and parquet formats are supported.")
-
-        return df
-
-    def ls(self, path: str = None) -> List[str]:
+    def cp(self, paths: List[str], from_path: str, to_path: str):
         """
-        Returns a list of files in a path.
+        Copies the contents of `from_path` to `to_path`.
 
         Args:
-            path (str, optional): Path to a folder. Defaults to None.
+            paths (List[str]): List of S3 objects paths (e.g. [s3://bucket/dir0/key0, s3://bucket/dir0/key1]).
```

Comment:
TBH I don't understand, why is there a 3rd arg if there's already "from" and "to"?

Reply:
It allows to copy files keeping subfolders structure, e.g. the following parameters will allow to copy cat=1/file1.parquet and cat=2/file2.parquet from folder_a to folder_b without a need of running the code twice
```
paths = [
's3://bucket-name/folder_a/cat=1/file1.parquet',
's3://bucket-name/folder_a/cat=2/file2.parquet'
]

from_path = 's3://bucket-name/folder_a'
to_path = 's3://bucket-name/folder_b'
```
In the result new files location will be as follows:
```
's3://bucket-name/folder_b/cat=1/file1.parquet'
's3://bucket-name/folder_b/cat=2/file2.parquet'
```

Reply:
Can you add this to the docstring and add some validation maybe, since I think the logic is that user has to provide either (paths and to_path) or (from_path and to_path)?

Reply:
All 3 arguments are (unfortunately) mandatory in the awswrangler `copy_objects()` method. 

Example of usage added as per [📝 added example to the cp() docstring](https://github.com/dyvenia/viadot/pull/579/commits/e0f8ed481867a3960ebd8cd103c4f8f2e99fe88e)

Reply:
So if I want to copy a folder with 100 subfolders, I need to specify all 100 of them in `paths`?

Reply:
Ah, I noticed you switched to `awswrangler` from `s3fs`, I think `s3fs` handles things better, and also has a benefit of a standard filesystem API. Do we actually need `awswrangler` for anything here?

Reply:
Agreed. Moved cp() method from awswrangler to s3fs as per [⚡️ switched cp() method from awswrangler to s3fs](https://github.com/dyvenia/viadot/pull/579/commits/44d79546e4f018a2f95716977ea50fbb2101073d)